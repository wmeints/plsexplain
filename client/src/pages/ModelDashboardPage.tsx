import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { State } from '../redux/store';
import * as actions from '../redux/actions';
import FeatureImportance from '../components/FeatureImportance';
import LoadingIndicator from '../components/LoadingIndicator';
import InteractivePlotCard from '../components/InteractivePlotCard';

const ModelDashboardPage = (): React.ReactElement => {
  const dispatch = useDispatch();
  const { data, layout } = useSelector((state: State) => state.modelExplanations.featureImportance);
  const featureProfile = useSelector((state: State) => state.modelExplanations.featureProfile);
  const {
    loadingFeatureImportance,
    loadingFeatureProfile,
  } = useSelector((state: State) => state.modelExplanations);

  useEffect(() => {
    dispatch(actions.fetchFeatureImportance());
  }, []);

  const loadFeatureProfile = (feature: string) => {
    dispatch(actions.fetchFeatureProfile({ name: feature }));
  };

  return (
    <div className="container">
      <div className="row">
        <div className="col">
          <h1 className="h4 mb-4 mt-4">Model explanations</h1>
        </div>
      </div>
      <div className="row">
        <div className="col">
          {loadingFeatureImportance && <LoadingIndicator text="Loading data..." />}
          {!loadingFeatureImportance && (
            <FeatureImportance
              className="mb-3"
              data={data}
              layout={layout}
              onFeatureSelected={(feature) => loadFeatureProfile(feature)}
            />
          )}
          <h2 className="h4 mt-4 mb-4">
            Feature profile
          </h2>
          <InteractivePlotCard
            plot={featureProfile}
            loadingText="Loading feature profile..."
            missingDataText="Please select a feature from the feature importance plot to see its profile."
            loading={loadingFeatureProfile}
          />
        </div>
      </div>
    </div>
  );
};

export default ModelDashboardPage;
