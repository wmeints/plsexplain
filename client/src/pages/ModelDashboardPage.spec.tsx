import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import fetchMock from 'jest-fetch-mock';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ModelDashboardPage from './ModelDashboardPage';
import featureImportance from '../__samples__/feature-importance.json';

const mockStore = configureStore([
  thunk,
]);

beforeEach(() => {
  fetchMock.resetMocks();
});

test('renders correctly', () => {
  const store = mockStore({
    modelExplanations: {
      loadingFeatureImportance: true,
      featureImportance,
    },
  });

  const wrapper = mount(
    <Provider store={store}>
      <ModelDashboardPage />
    </Provider>,
  );
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly with data loaded', () => {
  const store = mockStore({
    modelExplanations: {
      loadingFeatureImportance: false,
      featureImportance,
    },
  });

  const wrapper = mount(
    <Provider store={store}>
      <ModelDashboardPage />
    </Provider>,
  );

  expect(wrapper).toMatchSnapshot();
});
