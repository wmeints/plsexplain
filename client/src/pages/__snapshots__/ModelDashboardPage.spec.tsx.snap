// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ReactWrapper {}`;

exports[`renders correctly with data loaded 1`] = `ReactWrapper {}`;
